django/db/utils.py

```diff
@@ -276,22 +279,41 @@ def allow_relation(self, obj1, obj2, **hints):
                     return allow
         return obj1._state.db == obj2._state.db
 
-    def allow_migrate(self, db, model, **hints):
+    def allow_migrate(self, db, app_label, **hints):
         for router in self.routers:
             try:
                 method = router.allow_migrate
             except AttributeError:
                 # If the router doesn't have a method, skip to the next one.
-                pass
+                continue
+
+            argspec = inspect.getargspec(router.allow_migrate)
+            if len(argspec.args) == 3 and not argspec.keywords:
+                warnings.warn(
+                    "The signature of allow_migrate has changed from "
+                    "allow_migrate(self, db, model) to "
+                    "allow_migrate(self, db, app_label, model_name=None, **hints). "
+                    "Support for the old signature will be removed in Django 2.0.",
+                    RemovedInDjango20Warning)
+                allow = method(db, hints.get('model'))
```

Comment:
Here we could be extra smart and return `True` if `hints.get('model') is None`; that would restore the previous behaviour of RunPython/RunSQL not being routable when using an old router. In that case we'd need a second deprecation warning.